internal/plugin/lifecycle/lifecycle.go

```diff
@@ -134,6 +134,8 @@ func (impl Implementation) reconcileMetadata(
 		"primary", cluster.Status.CurrentPrimary,
 		"resources", sidecarContainer.Resources)
 
+	// migrate to InjectPluginInitContainerSidecarSpec (means dropping support for k8s < 1.29)
```

Comment:
I tested using the init container and experienced some issues. Also the fact that this plugin doesn't need to start before postgres, and could potentially delay the start times even further, kept me from pursuing it.